File: backend/src/main/java/com/myerasmusjourney/backend/dto/ExperienceDTO.java
package com.myerasmusjourney.backend.dto;

import com.myerasmusjourney.backend.enumeration.Category;

import java.time.LocalDate;

public record ExperienceDTO (
        Long id,
        LocalDate date,
        Float rating,
        String title,
        String description,
        Category category,
        CitySimpleDTO city,
        UserSimpleDTO author
){}

File: frontend/shared/src/models/ExperienceDTO.ts
import type { CitySimpleDTO } from "./CitySimpleDTO";
import type { UserSimpleDTO } from "./UserSimpleDTO";

export interface ExperienceDTO{
    id: number;
    title: string;
    date: string;
    rating: number;
    description: string;
    categories: string[];
    city: CitySimpleDTO;
    author: UserSimpleDTO;
}